src/main/java/com/incognia/common/ReasonCodeConstants.java

```diff
@@ -0,0 +1,55 @@
+package com.incognia.common;
+
+/**
+ * This Class objective is to help developers use the values present in the code field on the
+ * Reasons class.
+ */
+public class ReasonCodeConstants {
```

Comment:
What about using an enum with a string property?

Reply:
i personally think it makes it harder to use
it would be ReasonCodeConstants.EnumName.getMessage()? or are you suggesting creating the enum from the string we get in the api? i think both are worse, but i would like @ggml1 @figueredo input on this

Reply:
The advantage of an enum is that we transform the strings into symbols, making the code, in general, less error-prone, e.g. it is easier to create an operation that handles the symbols as we don't need to (re-)validate the string, or it is not possible to use another string by mistake where you expect the symbol. But I agree that the need of something like `getMessage()` is worse and creating the enum from the response is not a good idea because we would break the client every time we release a new reason.

I believe the most common usage of the `Reason` and this class would be something like:

```
  if (ReasonCode.MPOS_FRAUD.equals(reason.getCode())) {
    // do something
  }
```

what about creating an enum and provide a method to test equality with a `String`?

```
public enum ReasonCode {
  MPOS_FRAUD("mpos_fraud"),
  ...;
  
  public boolean isSameAs(String code) {
    return this.code.equals(code);
  }
}
```

Thus, the usage would be kept similar to having the utility class:

```
  if (ReasonCode.MPOS_FRAUD.isSameAs(reason.getCode())) {
    // do something
  }
```